src/components/formreporttype.jsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from "react";
+import { Formik, Form, Field } from "formik";
+import Categories from "./services/categories.json";
+import * as Yup from 'yup';
+import ErrorMsg from "./errormsg"
+
+const getSubCategories = (categories, categoryId) => {
+	var cats = categories.find(category => category.id === categoryId);
+	return cats ? cats.types : [];
+};
+
+
+const ServiceRequestForm = props => {
+
+	const [categories] = useState(Categories);
+	const [subCategories, setSubCategories] = useState([]);
+
+	const handleServiceRequestChange = (changeEvent) => {
+		const { value } = changeEvent.currentTarget;
+		const subCategories = getSubCategories(categories, parseInt(value));
+		setSubCategories(subCategories);
+	};
+
+
+	return (
+
+		<React.Fragment>
+			<div className="container Container-bg">
+				<h4>How Can We Help?</h4>
+				<Formik
+					initialValues={{
+						requestType: '',
+						subRequestType: '',
+					}}
+					validationSchema={Yup.object().shape({
+						requestType: Yup.string().required('Request Category is required'),
+						subRequestType: Yup.string().required('Sub Category is required'),
+					})}
+
+					onSubmit={(values, { setSubmitting }) => {
+						setTimeout(() => {
+							alert(JSON.stringify(values, null, 2));
+							setSubmitting(false);
+						}, 400);
+					}}
+				>
+					{
+						(props) => {
+							const { values, isSubmitting, errors, touched } = props;
+
+							return (
+
+								<Form onSubmit={props.handleSubmit}>
+									<label htmlFor="requestType">Request Category</label>
+									<br />
+									<Field
+										component="select"
+										id="requestType"
+										name="requestType"
+										onChange={e => {
```

Comment:
this should be

```js
...
onChange={handleServiceRequestChange}
...
```

i can't find the exact reasoning via another reference, but calling a function the way you have on every change, creates a new instance of that function. If you pass in handle function, it has already been declared and doesn't keep creating functions.

Reply:
let me know this changes works for you